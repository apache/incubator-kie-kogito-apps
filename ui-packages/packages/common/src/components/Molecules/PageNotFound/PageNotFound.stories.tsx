import React from 'react';
import PageNotFound from './PageNotFound';

export default {
  title: 'Page not found'
};

export const defaultView = (args) => {
  return <PageNotFound {...args} />;
};

defaultView.args = {
  defaultPath: '/DomainExplorer',
  defaultButton: 'Go to domain explorer',
  location: {}
};
