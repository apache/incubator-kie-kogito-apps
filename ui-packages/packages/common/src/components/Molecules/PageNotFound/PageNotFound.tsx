import React, { useState } from 'react';
import {
  PageSection,
  Bullseye,
  EmptyState,
  EmptyStateIcon,
  EmptyStateVariant,
  Button,
  EmptyStateBody,
  Title
} from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import { Redirect } from 'react-router';

interface IOwnProps {
  defaultPath: string;
  defaultButton: string;
  location: any;
}

const PageNotFound: React.FC<IOwnProps> = props => {
  let prevPath;
  if (props.location.state !== undefined) {
    prevPath = props.location.state.prev;
  } else {
    prevPath = props.defaultPath;
  }

  const tempPath = prevPath.split('/');
  prevPath = tempPath.filter(item => item);

  const [isRedirect, setIsredirect] = useState(false);
  const redirectHandler = () => {
    setIsredirect(true);
  };
  return (
    <>
      {isRedirect && <Redirect to={`/${prevPath[0]}`} />}
      <PageSection variant="light">
        <Bullseye>
          <EmptyState variant={EmptyStateVariant.full}>
            <EmptyStateIcon
              icon={ExclamationCircleIcon}
              size="md"
              color="var(--pf-global--danger-color--100)"
            />
            <Title headingLevel="h1" size="4xl">
              404 Error: page not found
            </Title>
            <EmptyStateBody>This page could not be found.</EmptyStateBody>
            <Button variant="primary" onClick={redirectHandler}>
              {props.defaultButton}               
            </Button>
          </EmptyState>
        </Bullseye>
      </PageSection>
    </>
  );
};

export default PageNotFound;
