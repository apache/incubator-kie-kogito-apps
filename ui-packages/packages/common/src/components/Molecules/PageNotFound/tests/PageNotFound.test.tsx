import React from 'react';
import { shallow } from 'enzyme';
import PageNotFound from '../PageNotFound';
import { Button } from '@patternfly/react-core';

const props1 = {
  defaultPath: '/processInstances',
  defaultButton: '',
  location: {}
};

const props2 = {
  defaultPath: '/processInstances',
  defaultButton: '',
  location: {
    state: {
      prev: '/processInstances',
      description: 'some description',
      buttonText: 'button'
    }
  }
};

describe('PageNotFound component tests', () => {
  it('snapshot testing without location object', () => {
    const wrapper = shallow(<PageNotFound {...props1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot testing with location object', () => {
    const wrapper = shallow(<PageNotFound {...props2} />);
    expect(wrapper).toMatchSnapshot();
  });
  /* tslint:disable */
  it('redirect button click', () => {
    const wrapper = shallow(<PageNotFound {...props2} />);
    wrapper.find(Button).simulate('click');
    expect(wrapper.find('Redirect').props()['to']).toEqual('/processInstances');
  });
});
