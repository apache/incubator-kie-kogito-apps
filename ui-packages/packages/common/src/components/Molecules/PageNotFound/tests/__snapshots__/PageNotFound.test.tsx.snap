// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component tests snapshot testing with location object 1`] = `
<Fragment>
  <PageSection
    data-ouia-component-type="page-not-found"
    data-ouia-safe={true}
    variant="light"
  >
    <Bullseye>
      <EmptyState
        variant="full"
      >
        <EmptyStateIcon
          color="var(--pf-global--danger-color--100)"
          icon={[Function]}
        />
        <Title
          headingLevel="h1"
          size="4xl"
        >
          404 Error: page not found
        </Title>
        <EmptyStateBody>
          This page could not be found.
        </EmptyStateBody>
        <Button
          onClick={[Function]}
          variant="primary"
        />
      </EmptyState>
    </Bullseye>
  </PageSection>
</Fragment>
`;

exports[`PageNotFound component tests snapshot testing without location object 1`] = `
<Fragment>
  <PageSection
    data-ouia-component-type="page-not-found"
    data-ouia-safe={true}
    variant="light"
  >
    <Bullseye>
      <EmptyState
        variant="full"
      >
        <EmptyStateIcon
          color="var(--pf-global--danger-color--100)"
          icon={[Function]}
        />
        <Title
          headingLevel="h1"
          size="4xl"
        >
          404 Error: page not found
        </Title>
        <EmptyStateBody>
          This page could not be found.
        </EmptyStateBody>
        <Button
          onClick={[Function]}
          variant="primary"
        />
      </EmptyState>
    </Bullseye>
  </PageSection>
</Fragment>
`;
